src/Components/Artwork/Details/__tests__/PrimaryLabelLine.jest.tsx

```diff
@@ -0,0 +1,209 @@
+import { DateTime } from "luxon"
+import { PrimaryLabelLineQueryRenderer } from "Components/Artwork/Details/PrimaryLabelLine"
+import { createMockEnvironment, MockPayloadGenerator } from "relay-test-utils"
+import { createRef } from "react"
+import { act, render, screen } from "@testing-library/react"
+import { useDidMount } from "Utils/Hooks/useDidMount"
+import { useIntersectionObserver } from "Utils/Hooks/useIntersectionObserver"
+import { useSystemContext } from "System/Hooks/useSystemContext"
+import { useArtworkGridContext } from "Components/ArtworkGrid/ArtworkGridContext"
+
+jest.unmock("react-relay")
+
+jest.mock("System/Hooks/useSystemContext")
+jest.mock("Components/ArtworkGrid/ArtworkGridContext")
+jest.mock("Utils/Hooks/useDidMount")
+jest.mock("Utils/Hooks/useIntersectionObserver")
+
+describe("PrimaryLabelLine", () => {
+  let mockuseDidMount = useDidMount as jest.Mock
+  let mockUseIntersectionObserver = useIntersectionObserver as jest.Mock
+  let mockUseSystemContext = useSystemContext as jest.Mock
+  let mockUseArtworkGridContext = useArtworkGridContext as jest.Mock
+  let mockEnvironment
+
+  beforeEach(() => {
+    mockuseDidMount.mockImplementation(() => true)
+
+    let called = false
+    mockUseIntersectionObserver.mockImplementation(({ onIntersection }) => {
+      if (!called) {
+        onIntersection()
+        called = true
+      }
+      return { ref: createRef() }
+    })
+
+    mockEnvironment = createMockEnvironment()
+    mockUseSystemContext.mockImplementation(() => {
+      return {
+        relayEnvironment: mockEnvironment,
+      }
+    })
+    mockUseArtworkGridContext.mockImplementation(() => {
+      return {}
+    })
+  })
+
+  describe("before fetching partner offer", () => {
+    it("shows increased interest when it's passed in as the label", () => {
+      render(
+        <PrimaryLabelLineQueryRenderer
```

Comment:
This is non-blocking, but suggested: We don't need to test query renderers,  since they typically only render fragments. So just test the fragment, using our standard relay testing  tools (`setupTestRendererTL`). That allows us to avoid having to wire up networking manually. Everything can happen inside of the test helper.

Reply:
ah, just saw my other comment. Ignore 👍

Reply:
Yeah you had the similar suggestion above. I understand it's usually not needed but I intentionally wanted to cover the [2-step rendering](https://github.com/artsy/force/pull/14897#discussion_r1876793670) due to this new server-renders-first-and-clientside-replaces-later behavior, and I thought testing the query renderer is a direct way for that purpose.